### PR TITLE
profiles/prefix: reorder macos clang parents and override LDFLAGS

### PR DESCRIPTION
**Pull request by DarthGandalf**
This is alternative to https://github.com/gentoo/gentoo/pull/35250

Bug: https://bugs.gentoo.org/758167

@grobian do you prefer this duplication instead?

### CODE REVIEW

File: profiles/prefix/darwin/macos/features/clang-lld/make.defaults [hunk 1]
```diff
@@ -5,3 +5,8 @@ LD=ld64.lld
 
 # Several packages fail to build without this
 lt_cv_apple_cc_single_mod=yes
+
+# This is the sort of equivalent of -Wl,--as-needed
+# It's already set in profiles/prefix/darwin, but profiles/features/llvm
+# appends --as-needed to it, so here we set it back to the previous value
+LDFLAGS="-Wl,-dead_strip_dylibs"
```
Comment (grobian) [inline profiles/prefix/darwin/macos/features/clang-lld/make.defaults:12]: just to be sure, ld.lld doesn't support --as-needed on macOS targets?  (e.g. it behaves as ld64?)

Comment (DarthGandalf) [reply profiles/prefix/darwin/macos/features/clang-lld/make.defaults:12]: That's why it's called ld64.lld and not ld.lld

Comment (grobian) [reply profiles/prefix/darwin/macos/features/clang-lld/make.defaults:12]: scratch that, I see you override LD to ld64.lld.  So then it makes more sense to redefine LDFLAGS too.  I'd group them in your commit so it make sense.

This is ok, it's duplicating a bit, but not to much, and I think it's much more clean than the split-profile approach.  Thanks!

Comment (DarthGandalf) [reply profiles/prefix/darwin/macos/features/clang-lld/make.defaults:9]: Done

**Comment by grobian** [comment]
thanks

**Pull request closed without merging**